### DISCUSSION TITLE
Message Content Intent failing

### DISCUSSION THREAD

**Discussion by mgolden356** [Tyrrrz/DiscordChatExporter#944] [Q&A]
On channels which the bot has administrator access to, with Message Content Intent enabled in the developer portal, the bot fails to export some/all of a channel, with the error that the Message Content Intent isn't enabled. This is using the latest version: GUI 2.36.1 fails partway, that is on some channels that gives the error, it will download the first 209 messages before giving the error, but on CLI 2.36.1 (both Windows btw) it would fail totally without download.

**Comment by 96-LB** [comment]
Timely post, because one of my exports just now came up short and I wasn't sure why. It seems to fail on this message:
![image](https://user-images.githubusercontent.com/21089343/193478168-2ab12b56-1933-4841-9527-4bcbffc0fbd1.png)
I'd bet our heuristic for checking whether message content is enabled fails to handle this type of message.

**Reply by mgolden356** [reply]
Ah, the message that failed for me, the 210th one, is a sticker (a large animated looping emote), which is "No longer available".
![image](https://user-images.githubusercontent.com/32921157/193478367-16092718-fa4d-4924-8904-3362bb2bff5d.png)

Perhaps the user or the Discord or the rules allowed the posting of this sticker, but now it's not an available option for me within the Discord.

The thing that both of our failed messages have in common is that they deny the bot access for use, maybe because it's revoked (temporary invite code) or it was never useable by the bot in the first place (Nitro stickers).

**Reply by 96-LB** [reply]
I used v2.36 to export my message and check the result. I got this:
```json
{
      "id": "[REDACTED]",
      "type": "Default",
      "timestamp": "2021-12-24T17:15:49.442+00:00",
      "timestampEdited": null,
      "callEndedTimestamp": null,
      "isPinned": false,
      "content": "",
      "author": {
        "id": "[REDACTED]",
        "name": "[REDACTED]",
        "discriminator": "[REDACTED]",
        "nickname": "Triacontakai",
        "color": "#32CD32",
        "isBot": false,
        "avatarUrl": "[REDACTED]"
      },
      "attachments": [],
      "embeds": [],
      "stickers": [],
      "reactions": [],
      "mentions": []
    }
```
Which understandably looks entirely empty to DCE. I'll check the API to see if there's some way we can distinguish these messages.

**Reply by mgolden356** [reply]
If I understand you correctly, it works in the previous version, 2.36. Now that I look in the changelog, I can see why (if this is the case):
"Added a check which will trigger an error if the provided bot account does not have the message content intent enabled. Note, however, that this check is based on heuristics which may result in false negatives."
https://github.com/Tyrrrz/DiscordChatExporter/blob/master/Changelog.md
